wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1012,6 +1012,12 @@ impl super::Instance {
             );
         };
 
+        let is_mesa = cfg!(linux);
```

Comment:
Lets say `(linux || android) && !nvidia` and you can identify nvidia by the PCI vendor ID. If it isn't in auxil::db, add it.